staging/src/k8s.io/apiserver/pkg/admission/plugin/validatingadmissionpolicy/internal/generic/controller_test.go

```diff
@@ -406,7 +405,12 @@ func TestInformerList(t *testing.T) {
 		myController.Informer().Run(testContext.Done())
 	}()
 
-	require.True(t, cache.WaitForCacheSync(testContext.Done(), myController.HasSynced))
+	defer func() {
+		testCancel()
+		wg.Wait()
```

Comment:
@alexzielenski can you review the changes to this test? It appears to have been slightly broken before.

Reply:
...I made more extensive changes and left some TODOs for you (sorry!)

Reply:
test/cel changes + todos lgtm